src/App.js

```diff
@@ -10,40 +15,165 @@ class App extends React.Component {
+      restart: "",
+      dealButton: <button onClick={this.changeResult}>Deal</button>,
     };
   }
 
   dealCards = () => {
     // this.state.cardDeck.pop() modifies this.state.cardDeck array
-    const newCurrCards = [this.state.cardDeck.pop(), this.state.cardDeck.pop()];
+    const newCurrCard1 = this.state.cardDeck.pop();
+    const newCurrCard2 = this.state.cardDeck.pop();
+    this.state.player1.currCard = newCurrCard1;
+    this.state.player2.currCard = newCurrCard2;
+  };
+
+  compareCards = () => {
+    if (this.state.player1.currCard.rank === this.state.player2.currCard.rank) {
+      this.state.player1.result = "draw";
+      this.state.player2.result = "draw";
+    } else if (
+      this.state.player1.currCard.rank > this.state.player2.currCard.rank
+    ) {
+      this.state.player1.result = "win";
+      this.state.player1.score += 1;
+      this.state.player2.result = "loss";
+    } else {
+      this.state.player2.result = "win";
+      this.state.player2.score += 1;
+      this.state.player1.result = "loss";
+    }
+  };
+
+  changeResult = () => {
+    this.dealCards();
+    this.compareCards();
     this.setState({
-      currCards: newCurrCards,
+      round: this.state.round + 1,
+      player1: this.state.player1,
+      player2: this.state.player2,
     });
+    if (this.state.cardDeck.length === 0) {
+      this.genRestartButton();
+    }
   };
 
-  render() {
-    // You can write JavaScript here, just don't try and set your state!
+  exeRestart = () => {
+    console.log(this.state.player1.winGame);
+    this.setState({
+      cardDeck: makeShuffledDeck(),
+      round: 0,
+      player1: {
+        winGame: this.state.player1.winGame,
+        score: 0,
+        result: "",
+        currCard: {},
+      },
+      player2: {
+        winGame: this.state.player2.winGame,
+        score: 0,
+        result: "",
+        currCard: {},
+      },
+      restart: "",
+      dealButton: <button onClick={this.changeResult}>Deal</button>,
+    });
+  };
 
-    // You can access your current components state here, as indicated below
-    const currCardElems = this.state.currCards.map(({ name, suit }) => (
-      // Give each list element a unique key
-      <div key={`${name}${suit}`}>
-        {name} of {suit}
+  genRestartButton = () => {
+    let winner = "Both player";
+    if (this.state.player1.score > this.state.player2.score) {
+      winner = "Player 1";
+      this.state.player1.winGame += 1;
+      this.setState({ player1: this.state.player1 });
+    } else if (this.state.player1.score < this.state.player2.score) {
+      winner = "Player 2";
+      this.state.player2.winGame += 1;
+      this.setState({ player2: this.state.player2 });
+    }
+    const restartElem = (
+      <div>
+        <p>{winner} is the final winner! Do you want to play again?</p>
+        <button onClick={this.exeRestart}>Restart</button>
+        <button onClick={window.close}>Quit</button>
```

Comment:
closing the window is quite extreme for a React app. By right we aim to have a single page application, that does never face any interruptions. Can we think of another way to handle this?